### ISSUE TITLE
Incorrect pose estimation vision correction java example

### ISSUE THREAD

**Issue by PGgit08** [PhotonVision/photonvision#1634]
In the PhotonLib pose estimation java example, old vision measurements are being ignored and only the most recent measurement from `getAllUnreadResults()` is being added to the WPILib pose estimator:

- `getEstimatedGlobalPose()` method **(the issue is in this method)**: https://github.com/PhotonVision/photonvision/blob/e8efef476b3b4681c8899a8720774d6dbd5ccf56/photonlib-java-examples/poseest/src/main/java/frc/robot/Vision.java#L86-L114

- `SwerveDrivePoseEstimator` vision update call using the `getEstimatedGlobalPose()` method: https://github.com/PhotonVision/photonvision/blob/e8efef476b3b4681c8899a8720774d6dbd5ccf56/photonlib-java-examples/poseest/src/main/java/frc/robot/Robot.java#L64-L73

**Comment by PGgit08** [comment]
i forgot to add the bug label

**Comment by mcm001** [comment]
Yeah agreed that that example looks like an anti pattern. Good catch!

**Comment by gerth2** [comment]
Yee. The example design predates `getAllUnreadResults()`, was just never updated.

**Comment by gerth2** [comment]
Fyi some overlap with #1632
